pyproject.toml

```diff
@@ -0,0 +1,134 @@
+[project]
+
+authors = [{name = "Jeffrey Larson"}, {name = "Stephen Hudson"},
+            {name = "Stefan M. Wild"}, {name = "David Bindel"},
+            {name = "John-Luke Navarro"}]
+
+dependencies = ["pip>=24.3.1,<25", "setuptools>=75.1.0,<76", "numpy>=1.21,<3",
+                "psutil>=5.9.4,<7", "pydantic>=1.10.12,<3", "pyyaml>=6.0,<7", "tomli>=1.2.1,<3"]
+
+description = "A Python toolkit for coordinating asynchronous and dynamic ensembles of calculations."
+name = "libensemble"
+requires-python = ">=3.9"
+license = {file = "LICENSE"}
+readme = "README.rst"
+
+classifiers = [
+        "Development Status :: 5 - Production/Stable",
+        "Intended Audience :: Developers",
+        "Intended Audience :: Science/Research",
+        "License :: OSI Approved :: BSD License",
+        "Natural Language :: English",
+        "Operating System :: POSIX :: Linux",
+        "Operating System :: Unix",
+        "Operating System :: MacOS",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.9",
+        "Programming Language :: Python :: 3.10",
+        "Programming Language :: Python :: 3.11",
+        "Programming Language :: Python :: 3.12",
+        "Programming Language :: Python :: Implementation :: CPython",
+        "Topic :: Scientific/Engineering",
+        "Topic :: Software Development :: Libraries :: Python Modules",
+]
+version = "1.4.2+dev"
+
+[project.urls]
+Documentation = "https://libensemble.readthedocs.io/en/main/"
+Repository = "https://github.com/Libensemble/libensemble"
+Issues = "https://github.com/Libensemble/libensemble/issues"
+
+[build-system]
+build-backend = "setuptools.build_meta"
+requires = ["setuptools", "wheel"]
+
+[tool.setuptools.packages.find]
+where = ["."]
+include = ["libensemble*"]
+
+[tool.pixi.project]
+channels = ["conda-forge"]
+platforms = ["osx-arm64", "linux-64", "osx-64"]
+
+[tool.pixi.pypi-dependencies]
+libensemble = { path = ".", editable = true }
+
+[tool.pixi.environments]
+default = []
+dev = ["dev"]
+
+[tool.pixi.feature.dev.dependencies]
```

Comment:
Thinking lower bounds should be based on what we support. Or if we want to be latest, maybe make a comment to that effect.